tests/unit/amo/components/TestDownloadFirefoxButton.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+
+import DownloadFirefoxButton, {
+  DownloadFirefoxButtonBase,
+  mapStateToProps,
+} from 'amo/components/DownloadFirefoxButton';
+import { dispatchClientMetadata } from 'tests/unit/amo/helpers';
+import {
+  getFakeI18nInst,
+  shallowUntilTarget,
+  userAgents,
+} from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  function render({
+    userAgent = userAgents.firefox[0],
+    ...props
+  } = {}) {
+    const { store } = dispatchClientMetadata({ userAgent });
+
+    return shallowUntilTarget(
+      <DownloadFirefoxButton
+        i18n={getFakeI18nInst()}
+        store={store}
+        {...props}
+      />,
+      DownloadFirefoxButtonBase,
+    );
+  }
+
+  it('renders nothing if the browser is Firefox Desktop', () => {
+    const root = render();
+
+    expect(root.find('.DownloadFirefoxButton')).toHaveLength(0);
+  });
+
+  it('renders nothing if the browser is Firefox for Android', () => {
+    const root = render({ userAgent: userAgents.firefoxAndroid[0] });
```

Comment:
Since you're using `shallowUntilTarget()` you can test all of this more like a real production app, i.e. with state changes:

```js
const { store } = dispatchClientMetadata({
  userAgent: userAgents.firefoxAndroid[0],
});
const root = render({ store });
```

After making that change, you can delete the `mapStateToProps()` tests because they will be redundant.

Reply:
That's actually what these tests are doing; `userAgent` is passed to a `store` that's passed to the component. But that's not obvious from the way these tests are written; I'll tidy it up and make it more obvious.

Reply:
I figured it's good to have extra tests for `mapStateToProps` but let me know if you still want them removed. I guess it's less code, but without testing them discretely it just feels like they're only ever tested indirectly.

Reply:
But fair enough actually it adds nothing. I'll clear it out.